### DISCUSSION TITLE
Empty `task.config.variables_to_fetch` when using `**kwargs`

### DISCUSSION THREAD

**Discussion by sergiobuj** [camunda-community-hub/pyzeebe#467] [Q&A] [answered]
Hello!
I am trying to understand the reason why the `task.config.variables_to_fetch` attribute is empty when including a `**kwargs` as a parameter.

Consider the following task definitions:

```python 
from pyzeebe import ZeebeWorker, create_insecure_channel

channel = create_insecure_channel(hostname="localhost", port=26500)
worker = ZeebeWorker(channel)

@worker.task(task_type="example")
def example_task_var(input: str) -> dict:
    return {"output": f"Hello world, {input}!"}

@worker.task(task_type="extended_example")
def example_task_var_and_kwargs(input: str, **kwargs) -> dict:
    return {"output": f"Hello world, {input}!"}
```

From these, I would expect the following:
```python
# This is fine and expected
assert task1.config.variables_to_fetch == ["input"], task1.config.variables_to_fetch

# Here, while we can't specify all the possible variables. We do know about 'input'.
assert task2.config.variables_to_fetch == ["input"], task2.config.variables_to_fetch
```

However, it returns empty:
```
assert task2.config.variables_to_fetch == ["input"], task2.config.variables_to_fetch
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: []
```

### Any ideas on what might be reasoning behind this choice?
Here's the source code: https://github.com/camunda-community-hub/pyzeebe/blob/9aa314da951a1445f5924cdf24af2d33d38e9ddd/pyzeebe/function_tools/parameter_tools.py#L10

Thanks!

**Comment by dimastbk** [comment, marked as answer]
Hi. It's empty because of pyzeebe retrieve all variables when `kwargs` is present (and `variables_to_fetch` is passed to [ActivateJobsRequest](https://docs.camunda.io/docs/8.3/apis-tools/grpc/#input-activatejobsrequest), because of empty `fetchVariable` retrieves all variables). See https://github.com/camunda-community-hub/pyzeebe/blob/master/pyzeebe/function_tools/parameter_tools.py#L10
